### ISSUE TITLE
Issue with Tensorflow Session

### ISSUE THREAD

**Issue by ducva** [CleanCut/green#163]
I created sample project at

[green_tensorflow_bug Repository](https://github.com/ducva/green_tensorflow_bug)

For now, I can't use green to run my unit test with Tensorflow.

# Run test with green:
`green test.py`

```,output
$ green test.py
Before load graph
Before create session
After creating session

<STUCK HERE>

```

# Run test with python

`python test.py`
```, output
$ python test.py
Before load graph
Before create session
After creating session
Test data: [[ 0.   0.5  0.   0.   0.   0.   0.   0.   2.   0.   0.   0.   0. ]]
Prepare to run Tensorflow Session
Finish run Tensorflow session
.
----------------------------------------------------------------------
Ran 1 test in 0.012s

OK
```

Please help!

**Comment by CleanCut** [comment]
TL;DR Problem caused by module running code on import that doesn't play well with multiple instances of itself running.  [Suggested fix in PR here to not run that code at import time.](https://github.com/ducva/green_tensorflow_bug/pull/1).

**Comment by MinchinWeb** [comment]
When I first saw this, I wondered if it was something with threads. Does running green on a single thread fix this?
```
green -s 1 test.py
```

**Comment by CleanCut** [comment]
@MinchinWeb No, because that only means a single _worker_ process.  You still have the main process that imports all the libraries to decide how they should be distributed to workers, and since the bug was caused by run-at-import issues, the import in the main process and the import of any worker process (including just a single one) would collide.

There are other classes of behavior problems that can be avoided by just using a single worker process, but not this particular one.

**Comment by MinchinWeb** [comment]
@CleanCut Thanks for the explanation!

**Comment by CleanCut** [comment]
Any time. 😎
